### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by jrhee17**
com.fasterxml.jackson:jackson-bom: 2.13.1 -> 2.13.2
com.linecorp.armeria:armeria-bom: 1.14.1 -> 1.15.0
io.micrometer:micrometer-bom: 1.8.2 -> 1.8.4
ch.qos.logback:logback-classic: 1.2.10 -> 1.2.11
com.github.node-gradle:gradle-node-plugin: 3.1.1 -> 3.2.1
com.google.guava: 31.0.1-jre -> 31.1-jre
com.puppycrawl.tools:checkstyle: 9.2.1 -> 10.1
net.javacrumbs.json-unit: 2.28.0 -> 2.32.0'
org.awaitility.awaitility: 4.1.1 -> 4.2.0
org.hibernate.validator:hibernate-validator: 6.1.6.Final -> 6.2.3.Final
org.mockito: 4.3.1 -> 4.4.0
org.slf4j: 1.7.35 -> 1.7.36
org.springframework.boot: 2.6.3 -> 2.6.5

### CODE REVIEW

File: dependencies.yml [hunk 1]
```diff
@@ -3,16 +3,16 @@
 #     If its classes are exposed in Javadoc, update offline links as well.
 #
 boms:
-- com.fasterxml.jackson:jackson-bom:2.13.1
-- com.linecorp.armeria:armeria-bom:1.14.1
-- io.micrometer:micrometer-bom:1.8.2
+- com.fasterxml.jackson:jackson-bom:2.13.2.20220324
```
Comment (ikhoon) [inline dependencies.yml:6]: Should we align the Jackson version with Armeria that depends on `2.13.2`?

File: dependencies.yml [hunk 2]
```diff
@@ -114,6 +114,7 @@ com.linecorp.armeria:
     - https://www.javadoc.io/doc/com.linecorp.armeria/armeria-javadoc/1.14.1/
 
 com.puppycrawl.tools:
+  # Requires jdk 11
   checkstyle: { version: '9.2.1' }
```
Comment (ikhoon) [inline dependencies.yml:118]: Should we disable `checkstyle` task if the build JDK is 8?
https://github.com/line/centraldogma/blob/2ee824a003d1185643a4eadd6dddecfc336fca97/.github/workflows/gradle.yml#L51

**Comment by jrhee17** [comment]
```
./gradlew dependencyUpdates
...
The following dependencies have later release versions:
 - com.bmuschko:gradle-docker-plugin [6.7.0 -> 7.3.0]
 - com.cronutils:cron-utils [5.0.5 -> 9.1.6]
     http://cron-parser.com/
 - com.fasterxml.jackson:jackson-bom [2.13.2 -> 2.13.2.20220324]
     https://github.com/FasterXML/jackson-bom
 - com.github.ben-manes.caffeine:caffeine [2.9.1 -> 3.0.6]
     https://github.com/ben-manes/caffeine
 - com.guardsquare:proguard-gradle [7.2.0 -> 7.2.1]
     https://www.guardsquare.com/proguard
 - com.jayway.jsonpath:json-path [2.2.0 -> 2.7.0]
     https://github.com/jayway/JsonPath
 - org.apache.curator:curator-recipes [4.3.0 -> 5.2.1]
     http://curator.apache.org
 - org.apache.curator:curator-test [4.3.0 -> 5.2.1]
     http://curator.apache.org
 - org.apache.shiro:shiro-core [1.3.2 -> 1.9.0]
     https://shiro.apache.org/
 - org.apache.thrift:libthrift [0.9.3 -> 0.16.0]
     http://thrift.apache.org
 - org.apache.zookeeper:zookeeper [3.5.8 -> 3.8.0]
     http://zookeeper.apache.org
 - org.eclipse.jgit:org.eclipse.jgit [5.13.0.202109080827-r -> 6.1.0.202203080745-r]
     ${jgit-url}
 - org.eclipse.jgit:org.eclipse.jgit.ssh.jsch [5.13.0.202109080827-r -> 6.1.0.202203080745-r]
     ${jgit-url}
 - org.hibernate.validator:hibernate-validator [6.2.3.Final -> 7.0.4.Final]
     http://hibernate.org/validator

Gradle release-candidate updates:
 - Gradle: [6.8.3 -> 7.4.1]
```

**Comment by jrhee17** [comment]
https://github.com/Guardsquare/proguard/issues/231

Unsure why armeria side passed, but let me leave `proguard` without updates

**Review by ikhoon — APPROVED** [review]
Thanks, @jrhee17!

**Comment by ikhoon** [comment]
Note: Flaky tests really harass the CI results. We may need to split flaky tests from core tests and add separate builds like we did in Armeria.

**Pull request merged**